pkgs/by-name/ps/pshash/package.nix

```
@@ -0,0 +1,31 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+  ghc,
+}:
+stdenv.mkDerivation {
+  pname = "pshash";
+  version = "0.1.11.0";
+  src = fetchFromGitHub {
+    owner = "thornoar";
+    repo = "pshash";
+    rev = "c671983699a3016793d78048fa72cc2557d101c2";
+    sha256 = "sha256-L7ZopLwmvrYy35sq6orxQp7tggjtDG2ueIaei+8+aq4=";
+  };
+  nativeBuildInputs = [ ghc ];
+  buildPhase = ''
+    ghc app/Main.hs -no-keep-o-files -no-keep-hi-files -o pshash
+  '';
+  installPhase = ''
+    mkdir -p $out/bin
+    mv pshash $out/bin/
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall

    install -Dm755 pshash -t $out/bin

    runHook postInstall
  '';
```
See https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases. Also do the same for `buildPhase`

Try using [`haskellPackages.mkDerivation`](https://nixos.org/manual/nixpkgs/stable/#haskell) instead to reduce boilerplate code